assets/app/view/game/route_selector.rb

```diff
@@ -258,31 +276,70 @@ def actions(render_halts)
           store(:routes, @routes)
         end
 
+        add_train = lambda do
+          @step.add_train(@routes)
+          store(:routes, @routes)
+        end
+
+        remove_train = lambda do
+          @step.remove_train(@selected_route)
+          store(:routes, @routes)
+        end
+
+        increase_train = lambda do
+          @step.increase_train(@selected_route)
+          store(:routes, @routes)
+        end
+
+        decrease_train = lambda do
+          @step.decrease_train(@selected_route)
+          store(:routes, @routes)
+        end
+
         submit_style = {
           minWidth: '6.5rem',
           marginTop: '1rem',
           padding: '0.2rem 0.5rem',
         }
 
-        revenue = begin
-          @game.format_revenue_currency(@game.routes_revenue(active_routes))
-        rescue Engine::GameError
-          '(Invalid Route)'
-        end
+        revenue = if @step.respond_to?(:total_str)
```

Comment:
too much garbage can we avoid?

Reply:
Yes, I think I need to create a new view for track_conversion.